### DISCUSSION TITLE
E450, M-Bus Hardware (BKW)

### DISCUSSION THREAD

**Discussion by cavalon** [scs/smartmeter-datacollector#23] [General]
Hi there

Great project, highly appreciated!

As I was struggling a little on how to set-up the hardware site, my notes in the following (and maybe helpful for others):

- Meter: E450 from a BWK reseller
- Cheap USB to M-Bus-Slave dongle: https://www.aliexpress.com/item/32894249052.html  
- Raspberry Pi Zero 2 W
- I used a 6-Pin RJ12 cable. Only the 2 wires in the middle are used (pin 3 & pin 4, counting from the left side)
- Pin 3 is then connected to the M-Bus dongle on the side where the capacitor is , pin 4 to the other side

Then, using the demo-image, everything worked out-of-the box!

Thanks again!

**Comment by raymar9** [comment]
Hi cavalon
Thank you for your feedback and notes about the hardware setup. I just updated our Wiki with some of your information (https://github.com/scs/smartmeter-datacollector/wiki).
Best regards

**Comment by aschlapbach** [comment]
How to access the E450 physically?

My EKZ installation is sealed, and I do not expect EKZ to accept me breaking the seal.

Is the RJ12 connector below the sealed cover?
(I assume that the type of E450 EKZ is using has no RJ12 connector)
![image](https://user-images.githubusercontent.com/66993559/232237302-669442f2-a06c-4165-bd44-2bdacacc521d.png)

Can I access the M-bus directly (without the RJ12)? If so, where? (I assume there are pins 28+29)
![image](https://user-images.githubusercontent.com/66993559/232237085-7aa9a8ab-ba32-4fb6-a9e7-16dce08fa5e0.png)
![image](https://user-images.githubusercontent.com/66993559/232237097-33f6051f-efb3-434f-85d5-f057ed90e285.png)

> (pin 3 & pin 4, counting from the left side)
> Pin 3 is then connected to the M-Bus dongle on the side where the capacitor is , pin 4 to the other side

To my understanding, polarity does not matter with M-bus.

Does EKZ need to activate the interface? Do I need a key? I saw the following comment [here](https://www.photovoltaikforum.com/thread/137994-landis-gyr-e450-auslesen-wie-ich-es-mache/):

"**Aktivierung**: Zu allererst muss man beim Netzbetreiber anrufen und die "Kundenschnittstelle" aktivieren lassen. Ich wurde gefragt ob ich
- die optische oder kabelgebundene (RJ12 bzw. MBus) Schnittstelle nutzen möchte => RJ12
- die Daten verschlüsselt oder unverschlüsselt haben möchte => unverschlüsselt"

**Reply by cavalon** [reply]
My E450 looks a little different:

![image](https://user-images.githubusercontent.com/6121398/232337953-d5ed31b2-4bb4-417a-a66f-cfebd563d99f.png)

I have not researched on how to connect the meter differently than via RJ12. To access the pins you need to open the cover / cut the seal - I guess...

If I remember correctly a technician from my power provider had to enable the interface (non-encrypted) on site. He also cut the seal from the RJ12 cover (the one now "hanging" down, this seal "made no sense" (his words)).

**Comment by raymar9** [comment]
Hi @aschlapbach
You need to contact EKZ and request the activation of the CII (consumer information interface) for local data readout. Do not break the seal yourself. EKZ will replace the cover (lower part) of the meter and provide one with a RJ12 socket that is connected internally to the M-Bus pins (see image above from cavalon).
EKZ will further enable the CII in your meter and provide a decryption key for the data pushed by the meter. By default the CII is not enabled, so even if you open the cover by yourself and connect the M-Bus dongle you probably won't receive any data. EKZ only provides the CII with encrypted data.
